### PR TITLE
DeveloperGuide: Add implementation details for SaveState

### PR DESCRIPTION
**Pull request by tanjieling0**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,6 +162,48 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
...
+* `StateStorage#isStateStorageExists()` — Checks if the file storing the states exists.
+* `StateStorage#deleteStateStorage()` — Deletes the file storing the states.
+* `StateStorage#getLastCommand()` — Returns the last string that is present in the command box before it was last closed. If the file is found, the text in the file is loaded into the command box, else a new file is created and empty string is returned.
+* `StateStorage#getFilePath()` — Returns the file path of the file storing the states.
+* `StateStorage#getFilePathString()` — Returns the file path of the file storing the states as a string.
+* `StateStorage#getDirectoryPath()` — Returns the directory path of the file storing the states.
+
+Given below is an example usage scenario and how the save state feature behaves at each step.
+
+Step 1. The user launches the application. When the command box is instantiated, it calls the `StateStorage#getLastCommand()` method to get the last command that was present in the command box before it was last closed. The text in the file retrieved via `StateStorage#loadState()` and loaded into the command box.
```
Comment (shaunlxw) [inline docs/DeveloperGuide.md]: Possible typo in last sentence. "The text in the file _is_ retrieved..."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -158,6 +162,48 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
...
+
+**Note:** If the storage file is not found a new empty file is created.
+
+Step 2. The user changes the input in the command box. The `StateStorage#writeState(String state)` method is called to save the current state of the command box into the file.
+
+<puml src="diagrams/SaveStateActivityDiagram.puml" alt="SaveStateActivityDiagram" />
+
+<box type="info" seamless>
```
Comment (shaunlxw) [inline docs/DeveloperGuide.md]: This box is not closed below. Can consider removing this line.

**Review by shaunlxw — CHANGES_REQUESTED** [review]
Very clear and detailed explanation 👍 Some minor changes required.

NIT:
1. The use of the word "component" for `StateStorage` might be inappropriate. In the DG, under section Design -> Architecture, it mentions the four main components. Seems like `StateStorage` falls under the `Storage` component.
2. Is there a way to integrate the `StateStorage` class into the class diagram under the "Storage component" subsection?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tanjieling0) [reply docs/DeveloperGuide.md]: removed

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (tanjieling0) [reply docs/DeveloperGuide.md]: thank you for pointing it out, fixed

**Comment by tanjieling0** [comment]
Screenshot 2024-03-29 at 3 40 49 PM
I have updated the UML diagrams and rephrased some of the implementation d
etails, but i have yet to implement the actual refactoring of the statestorage functions that I have indicated in the UML diagrams. Will submit a new PR with updated implementation after submission of DG draft!

**Review by purivirakarin — APPROVED** [review]
LGTM!

**Review by shaunlxw — APPROVED** [review]
LGTM! 👍

**Pull request merged**